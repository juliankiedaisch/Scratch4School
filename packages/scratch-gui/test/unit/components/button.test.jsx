import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom';
import ButtonComponent from '../../../src/components/button/button';

describe('ButtonComponent', () => {
    test('matches snapshot', () => {
        const {container} = render(
            <ButtonComponent onClick={jest.fn()} />
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    test('triggers callback when clicked', () => {
        const onClick = jest.fn();
        const {container} = render(
            <ButtonComponent onClick={onClick} />
        );

        const button = container.firstChild;

        fireEvent.click(button);

        expect(onClick).toHaveBeenCalledTimes(1);
    });

    test('does not trigger callback when not clicked', () => {
        const onClick = jest.fn();
        render(
            <ButtonComponent onClick={onClick} />
        );
        
        expect(onClick).toHaveBeenCalledTimes(0);
    });
});
