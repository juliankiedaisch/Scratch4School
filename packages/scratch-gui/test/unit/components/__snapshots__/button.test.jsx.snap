// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent matches snapshot 1`] = `
<span
  class=""
  role="button"
>
  <div />
</span>
`;
